File: src/types/customer.ts
// Customer/CRM Types

export interface CustomerContact {
  phone?: string;
  email?: string;
  address?: string;
  city?: string;
  state?: string;
  country?: string;
  pincode?: string;
}

export interface CustomerInteraction {
  id: string;
  customerId: string;
  agentId: string;
  agentName: string;
  type: 'visit' | 'call' | 'email' | 'meeting' | 'follow_up' | 'other';
  description: string;
  outcome?: 'positive' | 'neutral' | 'negative';
  nextFollowUp?: string; // ISO date string
  createdAt: string;
}

export interface CustomerNote {
  id: string;
  customerId: string;
  agentId: string;
  agentName: string;
  content: string;
  createdAt: string;
  updatedAt?: string;
}

export interface CustomerPurchase {
  id: string;
  customerId: string;
  agentId: string;
  agentName: string;
  productName: string;
  quantity: number;
  unitPrice: number;
  totalAmount: number;
  purchaseDate: string;
  notes?: string;
  createdAt: string;
}

export interface Customer {
  id: string;
  companyId: string;
  
  // Basic Info (from target or manual entry)
  name: string;
  businessName?: string;
  type: 'individual' | 'business';
  category?: string; // e.g., 'retail', 'wholesale', 'distributor'
  
  // Contact Info
  contact: CustomerContact;
  
  // Tags for categorization
  tags: string[];
  
  // Source tracking
  sourceType: 'target' | 'manual' | 'referral';
  sourceTargetId?: string; // If created from a target
  sourceAgentId?: string; // Agent who added this customer
  
  // Status
  status: 'active' | 'inactive' | 'prospect' | 'lead';
  priority: 'high' | 'medium' | 'low';
  
  // Location
  location?: {
    latitude: number;
    longitude: number;
  };
  
  // Stats
  totalInteractions: number;
  totalPurchases: number;
  totalPurchaseValue: number;
  lastInteractionDate?: string;
  lastPurchaseDate?: string;
  
  // Metadata
  createdAt: string;
  updatedAt: string;
  createdBy: string;
}

export interface CustomerFormData {
  name: string;
  businessName?: string;
  type: 'individual' | 'business';
  category?: string;
  contact: CustomerContact;
  tags: string[];
  status: 'active' | 'inactive' | 'prospect' | 'lead';
  priority: 'high' | 'medium' | 'low';
  location?: {
    latitude: number;
    longitude: number;
  };
}

export interface InteractionFormData {
  type: 'visit' | 'call' | 'email' | 'meeting' | 'follow_up' | 'other';
  description: string;
  outcome?: 'positive' | 'neutral' | 'negative';
  nextFollowUp?: string;
}

export interface PurchaseFormData {
  productName: string;
  quantity: number;
  unitPrice: number;
  purchaseDate: string;
  notes?: string;
}

export interface CustomerFilters {
  status?: string;
  priority?: string;
  category?: string;
  tags?: string[];
  searchQuery?: string;
  agentId?: string;
  dateFrom?: string;
  dateTo?: string;
}

export interface CustomerStats {
  totalCustomers: number;
  activeCustomers: number;
  prospects: number;
  leads: number;
  highPriority: number;
  totalInteractions: number;
  totalPurchaseValue: number;
  recentInteractions: number; // Last 7 days
}

File: android-native/app/src/main/java/com/poverse/app/data/model/Customer.kt
package com.poverse.app.data.model

import kotlinx.serialization.Serializable

@Serializable
data class Customer(
    val id: String = "",
    val name: String = "",
    val businessName: String = "",
    val type: CustomerType = CustomerType.INDIVIDUAL,
    val category: String = "",
    val email: String = "",
    val phone: String = "",
    val address: String = "",
    val city: String = "",
    val state: String = "",
    val country: String = "",
    val location: LocationPoint? = null,
    val tags: List<String> = emptyList(),
    val sourceType: String = "manual",
    val sourceId: String = "",
    val status: CustomerStatus = CustomerStatus.ACTIVE,
    val priority: TargetPriority = TargetPriority.MEDIUM,
    val totalInteractions: Int = 0,
    val totalPurchases: Double = 0.0,
    val lastInteractionDate: Long = 0,
    val companyId: String = "",
    val assignedTo: String = "",
    val notes: String = "",
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)

@Serializable
enum class CustomerType {
    INDIVIDUAL, BUSINESS;

    companion object {
        fun fromString(value: String): CustomerType = try {
            valueOf(value.uppercase())
        } catch (e: Exception) {
            INDIVIDUAL
        }
    }
}

@Serializable
enum class CustomerStatus {
    ACTIVE, INACTIVE, PROSPECT, LEAD;

    companion object {
        fun fromString(value: String): CustomerStatus = try {
            valueOf(value.uppercase())
        } catch (e: Exception) {
            ACTIVE
        }
    }

    fun displayName(): String = name.lowercase().replaceFirstChar { it.uppercase() }
}

@Serializable
data class CustomerInteraction(
    val id: String = "",
    val customerId: String = "",
    val userId: String = "",
    val userName: String = "",
    val type: InteractionType = InteractionType.VISIT,
    val description: String = "",
    val outcome: String = "",
    val createdAt: Long = System.currentTimeMillis()
)

@Serializable
enum class InteractionType {
    VISIT, CALL, EMAIL, MEETING, FOLLOW_UP, OTHER;

    companion object {
        fun fromString(value: String): InteractionType = try {
            valueOf(value.uppercase())
        } catch (e: Exception) {
            OTHER
        }
    }

    fun displayName(): String = name.lowercase().replace('_', ' ')
        .replaceFirstChar { it.uppercase() }
}

@Serializable
data class CustomerNote(
    val id: String = "",
    val customerId: String = "",
    val userId: String = "",
    val userName: String = "",
    val content: String = "",
    val createdAt: Long = System.currentTimeMillis()
)

@Serializable
data class CustomerPurchase(
    val id: String = "",
    val customerId: String = "",
    val userId: String = "",
    val productName: String = "",
    val quantity: Int = 0,
    val amount: Double = 0.0,
    val date: String = "",
    val notes: String = "",
    val createdAt: Long = System.currentTimeMillis()
)
